srcs/utils/free_memory.c

```diff
@@ -47,9 +47,9 @@ void	free_table(t_cmd_table *cmd_table, int nb_cmds)
 		if (cmd_table[i].tokens)
 			free_tokens(cmd_table[i].tokens, cmd_table[i].nb_tokens);
 		if (cmd_table[i].flags)
-			free_split(cmd_table[i].flags);
+			free(cmd_table[i].flags);

```

Comment:
Is this because you're freeing flags and args elsewhere?

Reply:
it's because they ARE tokens actually =) That's why we were getting the double free =) 
which is actually making me think I might be even giving memory to those two for no reason...

Reply:
FYI (that's the function where I'm handling flags and args.

```
static char	**get_args(t_token *tokens, int nb_tokens, t_token_type arg_type)
{
	int		i;
	int		j;
	int		nb_args;
	char	**args;

	i = 0;
	j = 0;
	nb_args = get_type_count(tokens, nb_tokens, arg_type);
	args = calloc_or_exit(sizeof(char*), nb_args + 1);
	while (i < nb_tokens)
	{
		if (tokens[i].type == arg_type)
			args[j++] = tokens[i].text;
		i++;
	}
	args[j] = 0;
	return (args);
}
```

Reply:
Ok I see, makes sense then. I think what you do is good, that you should indeed allocate memory for the flags and args